streamer/src/nonblocking/quic.rs

```diff
@@ -707,17 +711,30 @@ fn max_streams_for_connection_in_100ms(
     connection_type: ConnectionPeerType,
     stake: u64,
     total_stake: u64,
+    max_unstaked_connections: usize,
 ) -> u64 {
-    if matches!(connection_type, ConnectionPeerType::Unstaked) || stake == 0 {
+    let max_unstaked_streams_per_100ms = if max_unstaked_connections == 0 {
+        0
+    } else {
         Percentage::from(MAX_UNSTAKED_STREAMS_PERCENT)
             .apply_to(MAX_STREAMS_PER_100MS)
             .saturating_div(MAX_UNSTAKED_CONNECTIONS as u64)
+    };
+
+    let min_staked_streams_per_100ms = if max_unstaked_connections == 0 {
```

Comment:
`max_unstaked_connections == 0` please tell me that this isn't a sentinel for tpufwd port

Reply:
It is. But what it really means is also on that port we allow 0 unstaked connections and only staked ones. So literally it is correct. It does not matter if it is used for forward or not.

Reply:
i know what it means. that's not the problem. the problem is that it is extremely brittle logic that can be broken from other areas of the code base with no knowledge of its existence. we cannot have code like this